### DISCUSSION TITLE
Class testing in tests

### DISCUSSION THREAD

**Discussion by Alberth289346** [CorsixTH/CorsixTH#1767] [General Diagnostics]
Modified the objects to take a hospital parameter, see https://github.com/Alberth289346/CorsixTH/tree/hospital-argument-for-objects
Being a bit paranoid, I added
``` lua
assert(class.is(hospital, Hospital), "First argument is not a Hospital instance.")
```
as a check. This breaks in a modified Lua test:
``` text
Error → spec/entities/object_spec.lua @ 38
object.lua:  can create Object objects
../Lua/entities/object.lua:41: First argument is not a Hospital instance.
```
The changes:
``` diff
diff --git a/CorsixTH/Luatest/spec/entities/machine_spec.lua b/CorsixTH/Luatest/spec/entities/mac
hine_spec.lua
index 2e25ce4e..fc6885b0 100644
--- a/CorsixTH/Luatest/spec/entities/machine_spec.lua
+++ b/CorsixTH/Luatest/spec/entities/machine_spec.lua
@@ -28,6 +28,7 @@ require("entities/machine")
 
 describe("object.lua: ", function()
   local stub_world = {map = {}}
+  local stub_hospital = {world = stub_world}
   local tile_x, tile_y, direction = 10, 10, "west"
 
   local function createMachineWithFakeInput()
@@ -44,7 +45,7 @@ describe("object.lua: ", function()
       idle_animations = {west = true},
       orientations = {west = orientation}
       }
-    return Machine(stub_world, fake_object_type, tile_x, tile_y, direction)
+    return Machine(stub_hospital, fake_object_type, tile_x, tile_y, direction)
   end
 
   it("can update dynamic Info", function()
```

I guess the question is how to convince the test that the object gets a realistic fake hospital.

**Comment by TheCycoONE** [comment]
A few options but the preferred way would be to modify stub_hospital extend Hospital (setmetatable or use CorsixTH's class)
Alternately you could replace the global Hospital with your stub before the function was called.

**Comment by Alberth289346** [comment]
- Using metatables failed, the class check wasn't convinced (tried a few variations, however as I don't know how the class structure, I may have missed the one proper incantation).
- The `class "Hospital" {}` trick as described in the class.lua file gives a syntax error (due to lua 5.3 ?)
- Making an `Hospital` instance failed due to the class initializing the entire hospital which needs a game level, a map, etc etc
- Subclassing failed for the same reason (can't instantiate the thing)
- So hacking `_G["Hospital"]` was the only thing left and that worked.
